bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/ModuleUtil.java

```diff
@@ -0,0 +1,578 @@
+    }
+
+    /**
+     * Tries to convert raw value to its type.
+     *
+     * @param value raw value
+     * @param type the type to be converted to
+     * @return the converted value
+     * @throws Exception if conversion failed
+     */
+    private static Object convertToConfigType(String value, Type type) throws Exception {
+        Object result;
+        switch (type) {
+            case TEXT:
+                result = value;
+                break;
+            case BOOLEAN:
+                String trimmed = value.trim();
+                if (trimmed.equalsIgnoreCase("true")) {
+                    result = Boolean.TRUE;
+                } else if (trimmed.equalsIgnoreCase("false")) {
+                    result = Boolean.FALSE;
+                } else {
+                    result = null;
+                }
+                break;
+            case INTEGER:
+                result = Integer.valueOf(value);
+                break;
+            case DECIMAL:
+                result = Double.valueOf(value);
+                break;
+            default:
+                result = null;
+                break;
+        }
+        return result;
+    }
+
+    /**
+     * Validates if ConfigDescriptionParameter Type is equal to referred ConfigDescriptionParameter Type
+     *
+     * @param configDescription the source ConfigDescriptionParameter
+     * @param referredConfigName the referred ConfigDescriptionParameter name
+     * @param configDescriptionsMap Map with ConfigDescriptionParameter {@link #toConfigDescriptionMap(Set)}
+     */
+    private static void validateConfigType(ConfigDescriptionParameter configDescription, String referredConfigName,
+            Map<String, ConfigDescriptionParameter> configDescriptionsMap) {
+
+        Type configType = configDescription != null ? configDescription.getType() : null;
+        ConfigDescriptionParameter referredConfig = configDescriptionsMap.get(referredConfigName);
+        Type refferedConfigType = referredConfig != null ? referredConfig.getType() : null;
+        if ((configType == null || refferedConfigType == null) || !configType.equals(refferedConfigType)) {
+            throw new IllegalArgumentException("ConfigProperties Type missmatch:" + configDescription.getName() + ":"
+                    + configType + "->" + referredConfigName + ":" + refferedConfigType);
+        }
+    }
+
+    /**
+     * Validates source Input\Output value is assignableFrom Input\Output type.
+     *
+     * @param referredType the referred Input\Output type (fully qualified name i.e. java.lang.String)
+     * @param value the source Input\Output value
+     * @return true if is assignableFrom referred Input\Output type, false otherwise.
+     */
+    private static boolean validateConnectionType(String referredType, Object value) {
+        boolean valid = false;
+        try {
+            Class<?> clazzOutputExpected = FrameworkUtil.getBundle(ModuleUtil.class).loadClass((referredType));
```

Comment:
Hm, will this actually always work? If we define modules that should use org.eclipse.smarthome.core.items.Item, this won't be resolvable as there is no dependency to this package from the automation.core bundle.
We would imho have to use the class loader or the respective module type provider, don't you think?